ykpack/src/types.rs

```diff
@@ -133,14 +131,65 @@ pub struct PlaceProjection {
 /// Describes a projection operation upon a projection base.
 #[derive(Serialize, Deserialize, PartialEq, Eq, Debug, Clone)]
 pub enum ProjectionElem<V> {
```

Comment:
The parameterisation here irritates me. Especially since `ProjectionElem` is only ever instantiated with `V` as either `()` or `Local`. Toying with flattening it out...

Reply:
Projection elements are sub-views on aggregate structures.

So if we have `mystruct.f`, then `mystruct` is the projection "Base" and `f` is the "Element".

Similar notions for slices, indexing and derefs.

I'm going to have a play and see if we can come up with something a little more concise that what MIR does.

Reply:
[Taking a quick detour to get my upstream Rust change merged, because it will affect the TIR lowering]

Reply:
OK.